Diagnostic Cancer list update [DDP-6771]

bskinner opened the pull request:
DDP-6771

## Context
This branch includes the complete list of changes to the Cancer diagnostic list from CMI, as of **2021/08/11**

## FUD Score
- [x] :relaxed: All good, business as usual!

## How do we demo these changes?
The changes will be visible in `dev`, but must be deployed using `study-builder` first. The study can also be deployed locally for folks who want to check it out there.

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing
- [x] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release
- [x] These changes require no special release procedures--just code!

ssettipalli commented on study-builder/studies/pancan/subs.conf:
```diff
@@ -139,8 +139,16 @@
-      (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_SARCOMAS_S_LEIOMYO")
-      || user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_CHILD"].children["PRIMARY_CANCER_CHILD"].answers.hasOptionStartsWith("C_SARCOMAS_S_LEIOMYO"))
+      (
+        (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_SARCOMAS_S_LEIOMYO")
```
If its just 2 or 3 stableIds and they start with different string pattern like here we can use "hasAnyOption" rather than 3 pex with a  ||
For ex: here we can use:
(user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasAnyOption("C_GYNECOLOGIC_UTERINE_LEIOMYOSARCOMA", "C_SARCOMAS_S_LEIOMYO", "C_SARCOMA_CUTANEOUS_LEIMYOSARCOMA")

bskinner replied:
Thank you- that's helpful to know!

bskinner commented on study-builder/studies/pancan/snippets/common/cancer-picklist-groups-sarcomas.conf:
```diff
@@ -40,376 +42,388 @@
     },
     {
-      "stableId": "C_SARCOMAS_S_OSTEO_SARCOMA",
+      "stableId": "C_SARCOMAS_S_LEIOMYO_LMS_SARCOMA",
```
I didn't rename this stable id since it was modified in `develop` after I branched. Should be easy enough to go back and update after the fact, but it can be left as-is.

bskinner committed:
Remove the blood cancers group

bskinner committed:
Removes the additional cancers group

bskinner committed:
fixup: remove the c_blood translations

bskinner committed:
fixup: remove c_additional translations

bskinner committed:
LMS PEX redirect refactoring

bskinner committed:
PEX refactor for osteosarcoma changes

bskinner committed:
Removes deleted Gynegologic options

bskinner committed:
Removes deleted Brain & CNS options

bskinner committed:
Removes deleted options from the Breast group

bskinner committed:
Removes deleted options from the Endocrine group

bskinner committed:
Removes deleted Gastrointestinal options

bskinner committed:
Removes deleted genitourinary options

bskinner committed:
Removed deleted head & neck options

bskinner committed:
Removes deleted leukemia options

bskinner committed:
Removes deleted lymphoma options

bskinner committed:
Removes deleted lungs options

bskinner committed:
Removes deleted sarcoma options

bskinner committed:
Removes all deleted skin cancer options

bskinner committed:
Removes deleted thyroid cancer options

bskinner committed:
Removes deleted thymic cancer option

bskinner committed:
Updates Brain & CNS group to latest & greatest

bskinner committed:
Updates Gastrointestinal group to latest & greatest

bskinner committed:
New additions & updates for the Endocrine options

bskinner committed:
Additions & updates to genitourinary options

bskinner committed:
Additions & updates for the Gynecologic group

Includes changes to PEX for the Leiomyosarcoma Project redirects

bskinner committed:
Additions & updates for the Head & Neck group

bskinner committed:
Additions & updates for the Leukemia picklist group

bskinner committed:
Additions & updates for the Lymphomas picklist group

bskinner committed:
Additions & updates for the Lungs picklist group

bskinner committed:
Additions & updates for the melanomas picklist group

bskinner committed:
Additions & updates for the sarcomas picklist group

Includes changes for associated PEX expressions

bskinner committed:
Additions & updates for the skin picklist group

bskinner committed:
Additions & updates for the thymic picklist group

bskinner committed:
Additions & updates for the thyroid picklist group

bskinner committed:
Adds the MPNs picklist group

bskinner committed:
Adds the remaining ungrouped options

bskinner committed:
Removes the blood & additional cancer group includes

bskinner committed:
Stable IDs must be under 45 characters long

Renames c_leukemia_bpdcn_blastic_plasmacytoid_dendritic and c_lymphomas_pcfcl_primary_cutaneous_follicular

bskinner committed:
Variable names do not have a '$' prefix

bskinner committed:
Refactor to use hasAnyOption

bskinner committed:
Updated translations

ssettipalli commented on study-builder/studies/pancan/subs.conf:
```diff
@@ -143,8 +143,8 @@
     "has_lms": """
-      (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_SARCOMAS_S_LEIOMYO")
-      || user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_CHILD"].children["PRIMARY_CANCER_CHILD"].answers.hasOptionStartsWith("C_SARCOMAS_S_LEIOMYO"))
+      (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasAnyOption("C_SARCOMAS_S_LEIOMYO_LMS_SARCOMA", "C_GYNECOLOGIC_UTERINE_LEIOMYOSARCOMA", "C_SARCOMA_CUTANEOUS_LEIMYOSARCOMA")
```
looks good

ssettipalli reviewed (approved):
verified pex updates in subs.conf
 looks good to me.
